sdk/rust/src/processor/mod.rs

```diff
@@ -0,0 +1,144 @@
+                info!("sending TpRegisterRequest: {} {}",
+                      &handler.family_name(),
+                      &version);
+                let serialized = request.write_to_bytes().unwrap();
+                let x : &[u8] = &serialized;
+
+                let correlation_id: String = rand::thread_rng().gen_ascii_chars().take(16).collect();
+                let mut future = sender.send(
+                    Message_MessageType::TP_REGISTER_REQUEST,
+                    &correlation_id,
+                    x).unwrap();
+
+                // Absorb the TpRegisterResponse message
+                let _ = future.get().unwrap();
+            }
+        }
+
+        loop {
+            match receiver.recv() {
+                Ok(r) => {
+                    let message = r.unwrap();
+                    info!("Message: {}", message.get_correlation_id());
+
+                    match message.get_message_type() {
+                        Message_MessageType::TP_PROCESS_REQUEST => {
+                            let request: TpProcessRequest = protobuf::parse_from_bytes(&message.get_content()).unwrap();
+                            let mut context = TransactionContext::new(request.get_context_id(), sender.clone());
+
+                            let mut response = TpProcessResponse::new();
+                            match self.handlers[0].apply(&request, &mut context) {
+                                Ok(()) => {
+                                    response.set_status(TpProcessResponse_Status::OK);
+                                    info!("TP_PROCESS_REQUEST sending TpProcessResponse: OK");
+                                },
+                                Err(ApplyError::InvalidTransaction(msg)) => {
+                                    response.set_status(TpProcessResponse_Status::INVALID_TRANSACTION);
+                                    response.set_message(msg.clone());
+                                    info!("TP_PROCESS_REQUEST sending TpProcessResponse: {}", msg);
+                                },
+                                Err(err) => {
+                                    response.set_status(TpProcessResponse_Status::INTERNAL_ERROR);
+                                    response.set_message(String::from(err.description()));
+                                    info!("TP_PROCESS_REQUEST sending TpProcessResponse: {}", err.description());
+                                }
+                            };
+
+                            let serialized = response.write_to_bytes().unwrap();
+                            let x : &[u8] = &serialized;
+                            sender.reply(
+                                Message_MessageType::TP_PROCESS_RESPONSE,
+                                message.get_correlation_id(),
+                                x).unwrap();
+                        },
+                        _ => {
+                            let mut response = TpProcessResponse::new();
```

Comment:
I'm not sure what sort of behavior this will result in: sending back a response is good (i.e. the future on the other end will resolve), but it will potentially be the wrong type of message.

Reply:
Missing handling of `PING_REQUEST`, thought this could probably be done in a future PR

Reply:
I can include the Ping functionality with the reconnect work.

Reply:
We could also include the default handling rework in with reconnect work (probably some affinity to testing it all together).